ddl/ddl_worker.go

```diff
@@ -813,6 +813,21 @@ func (w *worker) waitSchemaSynced(d *ddlCtx, job *model.Job, waitTime time.Durat
 	w.waitSchemaChanged(ctx, d, waitTime, latestSchemaVersion, job)
 }
 
+func buildPlacementAffects(oldIDs []int64, newIDs []int64) []*model.AffectedOption {
+	if len(oldIDs) == 0 {
+		return nil
+	}
+
+	affects := make([]*model.AffectedOption, len(oldIDs))
+	for i := 0; i < len(oldIDs); i++ {
+		affects[i] = &model.AffectedOption{
```

Comment:
What about schemaID?

Reply:
It is not used anyway. Affects are processed by L141-165, then continue the loop. It is never passed to `ApplyDiff` or whatever function that required `schemaID`.